tests/test_envs.py

```diff
@@ -1,142 +1,134 @@
-
-
-def test_minecart():
-    env = mo_gym.make("minecart-v0")
-    env = LinearReward(env)
-    check_env(env, skip_render_check=True)
-    _test_pickle_env(env)
-
-
-def test_mountaincar():
-    env = mo_gym.make("mo-mountaincar-v0")
-    env = LinearReward(env)
-    check_env(env, skip_render_check=True)
-    _test_pickle_env(env)
-
-
-def test_continuous_mountaincar():
-    env = mo_gym.make("mo-mountaincarcontinuous-v0")
-    env = LinearReward(env)
+all_testing_env_specs = []
+for env_spec in gym.envs.registry.values():
+    if type(env_spec.entry_point) is not str:
+        continue
+    # collect MO Gymnasium envs
+    if env_spec.entry_point.split(".")[0] == "mo_gymnasium":
+        # Ignore highway as they do not deal with the random seed appropriately
+        if not env_spec.id.startswith("mo-highway"):
+            all_testing_env_specs.append(env_spec)
+
+
+@pytest.mark.parametrize(
+    "spec",
+    all_testing_env_specs,
+    ids=[spec.id for spec in all_testing_env_specs],
+)
+def test_all_env_api(spec):
+    """Check that all environments pass the environment checker."""
+    env = mo_gym.make(spec.id)
+    env = mo_gym.LinearReward(env)
     check_env(env, skip_render_check=True)
     _test_pickle_env(env)
 
 
-def test_resource_gathering():
-    env = mo_gym.make("resource-gathering-v0")
-    env = LinearReward(env)
-    check_env(env, skip_render_check=True)
-    _test_pickle_env(env)
+@pytest.mark.parametrize("spec", all_testing_env_specs, ids=[spec.id for spec in all_testing_env_specs])
+def test_all_env_passive_env_checker(spec):
+    env = mo_gym.make(spec.id)
+    env.reset()
+    env.step(env.action_space.sample())
+    env.close()
 
 
-def test_fruit_tree():
-    env = mo_gym.make("fruit-tree-v0")
-    env = LinearReward(env)
-    check_env(env, skip_render_check=True)
-    _test_pickle_env(env)
+# Note that this precludes running this test in multiple threads.
+# However, we probably already can't do multithreading due to some environments.
+SEED = 0
+NUM_STEPS = 50
 
 
-def test_mario():
-    env = mo_gym.make("mo-supermario-v0")
-    env = LinearReward(env)
-    check_env(env, skip_render_check=True)
-    _test_pickle_env(env)
+@pytest.mark.parametrize(
+    "env_spec",
+    all_testing_env_specs,
+    ids=[env.id for env in all_testing_env_specs],
+)
+def test_env_determinism_rollout(env_spec: EnvSpec):
+    """Run a rollout with two environments and assert equality.
+    This test run a rollout of NUM_STEPS steps with two environments
+    initialized with the same seed and assert that:
+    - observation after first reset are the same
+    - same actions are sampled by the two envs
+    - observations are contained in the observation space
+    - obs, rew, done and info are equals between the two envs
+    """
+    # Don't check rollout equality if it's a nondeterministic environment.
+    if env_spec.nondeterministic is True:
+        return
 
+    env_1 = env_spec.make(disable_env_checker=True)
```

Comment:
`mo_gym.make` ?

Reply:
mo_gym.make simply calls gym.make(disable_env_checker=True), so this is fine.